minpy/dispatch/policy.py

```diff
@@ -6,30 +6,55 @@
 from __future__ import print_function
 
 import functools
+import atexit
 import minpy
+from .rule import Blacklist
 from minpy.array import Value
 from minpy.array_variants import ArrayType
 from minpy.utils import log
 
 # pylint: disable= invalid-name
 _logger = log.get_logger(__name__)
-# pylint: enable= invalid-name
+
+# pylint: edle= invalid-name
 
 
 class PrimitivePolicyError(ValueError):
-    """Error during choosing primitives."""
-    pass
+    """Error during choosing primitives.
+
+    Parameters
+    ----------
+    name : str
+        Name waiting for dispatch.
+    policy_name : str
+        Name of the policy in which the error occurs.
+    """
+    def __init__(self, name, policy_name):
+        super(PrimitivePolicyError,
+              self).__init__("Cannot find implementation for function: %s() "
+                             "under policy: %s. Maybe lack of gradient "
+                             "implementation?", name, self.name)
+
 
 class Policy(object):
```

Comment:
nit: maybe we could change the name `Policy` to `Mode` since we don't have `PreferMXNetPolicy` right now.